### DISCUSSION TITLE
Speed up type coverage check

### DISCUSSION THREAD

**Discussion by pelmered** [pestphp/pest#1311] [Ideas]
The type coverage check is really great, but unfortunately it is also really slow. It would be really great if it could be faster. 

For my current project which is of medium size I would say (around 1200 files checked) it takes around 10 minutes to run the type coverage check in a GitHub Action. Locally on my Mac M1 Max it takes around 2.5 minutes. That is quite a lot slower than I would like it to be. 

Ideas to make it faster:
- Cache results and check only files that are changed. Could use either file checksum or maybe modified date so skip the check for that file. It is typically only a very small subset of the files that would be changed between runs. In GitHub Actions it should be possible to use `actions/cache` to restore this cache before you run.
- Support parallel processing of files. That should speed up the processing time significantly. The `--parallel` does not seem to do anything at the moment. 

Bonus request: It would also be cool if you could limit the check to specific folders/paths. Maybe also an option to only show files that does not have 100 %, or under a specified threshold.

**Comment by sertxudev** [comment]
I'm also waiting for `--parallel` option support on type coverage.
I have a GitHub runner with 24 processes, so all the parallel test are blazing fast except type coverage.

**Comment by nunomaduro** [comment]
this is fixed on pest v4, that will be out this summer.

**Reply by sertxudev** [reply]
Awesome! Thank you Nuno ❤

**Reply by pelmered** [reply]
That is great! Thank you! 👏
